python/external_config/remote_command.py

```diff
@@ -0,0 +1,288 @@
+        """
+        Serializes the current object into a string.
+
+        For use with :meth:`from_string`.
+
+        :returns: String representing the current instance.
+        :rtype: str
+        """
+        data = {
+            "GENERATION": CONFIGURATION_GENERATION,
+            "callback_name": self._callback_name,
+            "display_name": self._display_name,
+            "tooltip": self._tooltip,
+            "plugin_id": self._plugin_id,
+            "engine": self._engine,
+            "interpreter": self._interpreter,
+            "descriptor_uri": self._descriptor_uri,
+            "pipeline_config_id": self._pipeline_config_id,
+            "entity_type": self._entity_type,
+            "entity_id": self._entity_id,
+            "pipeline_config_name": self._pipeline_config_name
+
+        }
+        return cPickle.dumps(data)
+
+    @property
+    def pipeline_configuration_name(self):
+        """
+        The name of the Shotgun pipeline configuration this command is associated with,
+        or ``None`` if no association exists.
+        """
+        return self._pipeline_config_name
+
+    @property
+    def display_name(self):
+        """
+        Display name, suitable for display in a menu.
+        """
+        return self._display_name
+
+    @property
+    def tooltip(self):
+        """
+        Associated help text tooltip.
+        """
+        return self._tooltip
+
+    def execute(self):
+        """
+        Executes the remote command in a separate process.
+
+        .. note:: The process will be launched in an asynchronous way.
+            It is recommended that this command is executed in a worker thread.
+        """
+        # local imports becuase this is executed from runner scripts
+        from .process_execution import ProcessRunner
+        from .util import create_parameter_file
+
+        logger.debug("%s: execute command" % self)
+
+        script = os.path.abspath(
+            os.path.join(
+                os.path.dirname(__file__),
+                "scripts",
+                "external_runner.py"
+            )
+        )
+
+        args_file = create_parameter_file(
+            dict(
+                action="execute_command",
+                callback_name=self._callback_name,
+                configuration_uri=self._descriptor_uri,
+                pipeline_config_id=self._pipeline_config_id,
+                plugin_id=self._plugin_id,
+                engine_name=self._engine,
+                entity_type=self._entity_type,
+                entity_id=self._entity_id,
+                bundle_cache_fallback_paths=self._bundle.engine.sgtk.bundle_cache_fallback_paths,
+            )
+        )
+
+        args = [
+            self._interpreter,
+            script,
+            sgtk.bootstrap.ToolkitManager.get_core_python_path(),
+            args_file
+        ]
+        logger.debug("Command arguments: %s", args)
+
+        retcode, stdout, stderr = ProcessRunner.call_cmd(args)
```

Comment:
we discussed this - but it may be tricky because each process is likely to require it's own sgtk environment (or it *might* be possible with core swapping, but certainly safer this way). I think it's about finding the right tradeoff between performance and complexity. Let me do the launch publisher/run QT in the external process tests and then i think we will know more.